lib/maths/CBoostedTreeImpl.cc

```diff
@@ -1206,16 +1405,36 @@ void CBoostedTreeImpl::candidateRegressorFeatures(const TDoubleVec& probabilitie
     }
 }
 
-void CBoostedTreeImpl::refreshPredictionsAndLossDerivatives(core::CDataFrame& frame,
-                                                            const core::CPackedBitVector& trainingRowMask,
-                                                            const core::CPackedBitVector& testingRowMask,
-                                                            double eta,
-                                                            double lambda,
-                                                            TNodeVec& tree) const {
+void CBoostedTreeImpl::removePredictions(core::CDataFrame& frame,
+                                         const core::CPackedBitVector& trainingRowMask,
+                                         const core::CPackedBitVector& testingRowMask,
+                                         const TNodeVec& tree) const {
+
+    core::CPackedBitVector rowMask{trainingRowMask | testingRowMask};
 
-    using TArgMinLossVec = std::vector<CArgMinLoss>;
+    frame.writeColumns(m_NumberThreads, 0, frame.numberRows(),
+                       [&](TRowItr beginRows, TRowItr endRows) {
```

Comment:
This is the function signature expected for the callback. I agree we should probably be passing by constant reference. The types are lightweight and compared to the work we do in these calls copying relatively small objects here is probably negligible, but it is no more complex to pass by constant reference. Since I'd have to change every single call, and there are quite a few, let's make this change in a separate PR.